### DISCUSSION TITLE
Fastapi returns incorrent response

### DISCUSSION THREAD

**Discussion by smqizq** [fastapi/fastapi#8405] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import List
from fastapi import Depends, FastAPI
from sqlalchemy.orm import Session
from sqlalchemy import Column, String, Integer

from loguru import logger

from sqlalchemy import create_engine
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.orm import sessionmaker

# database
SQLALCHEMY_DATABASE_URL = "sqlite:///./sql_app.db"
engine = create_engine(
    SQLALCHEMY_DATABASE_URL, 
    connect_args={"check_same_thread": False},
)
SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
Base = declarative_base()
Base.metadata.create_all(bind=engine)

def get_db():
    db = SessionLocal()
    try:
        yield db
    finally:
        db.close()

# model
class Label(Base):
    __tablename__ = 'labels'
    id = Column(Integer, primary_key=True, nullable=False, index=True, unique=True, comment='id')
    name = Column(String(255), nullable=False)

    def __repr__(self):
        return f'{{ "id": "{self.id}": "name": "{self.name}" }}'

# schemas
from pydantic import BaseModel

class LabelBase(BaseModel):
    id: int

    class Config:
        orm_mode = True

class LabelSet(LabelBase):
    class Config:
            orm_mode = True

# app
app = FastAPI()

def set_label(db: Session, id: int):
    db_label = db.query(Label).filter(Label.id == id).first()
    db.commit()
    # db.refresh(db_label)
    # logger.info(db_label)
    return db_label

@app.post("/labels")
def set_labels(labels_to_set: List[LabelSet]=None, db: Session=Depends(get_db)):
    res = []
    for label_to_set in labels_to_set:
        db_label = set_label(db, label_to_set.id)
        res.append(db_label)
        logger.info(f"set_label: {db_label}") # this line always prints the correct label item
        # logger.info(res) # Add this line will return res: [{"name":"a","id":1},{"name":"b","id":2},{"name":"c","id":3}]
                       # Not add above line will retrun res: [{},{},{"name":"c","id":3}]
    return res

# curl test
# curl -X 'POST' \
#   'http://127.0.0.1:5000/labels' \
#   -H 'accept: application/json' \
#   -H 'Content-Type: application/json' \
#   -d '[
#   {
#     "id": 1
#   },
# {
#     "id": 2
#   },
# {
#     "id": 3
#   }
# ]'
```

### Description

* Run the above code 
* Run the curl test
* You'll find the response is incorrect, see the comment in code for detail.

### Operating System

Windows

### FastAPI Version

0.85.1

### Python Version

3.8.10

### Additional Context

for test, please insert the test data first:
[{"name":"a","id":1},{"name":"b","id":2},{"name":"c","id":3}]

**Comment by mbroton** [comment]
Hi! It would be great if you could be more precise - what's the expected and actual response?
I can see comments, but I'm not sure. I could be guessing, but I would rather not.

**Comment by smqizq** [comment]
> Hi! It would be great if you could be more precise - what's the expected and actual response? I can see comments, but I'm not sure. I could be guessing, but I would rather not.

Hi, the expected response is [{"name":"a","id":1},{"name":"b","id":2},{"name":"c","id":3}]
but actually returned [{},{},{"name":"c","id":3}]

**Comment by smqizq** [comment]
I Believe this is related to sqlalchemy. I'll look into it and update here

**Comment by tojimahammatov** [comment]
Hi, have you checked db_level before appending it to `res` list in a for loop ? Perhaps you prefer to use async def since you are making db connection, perhaps the result is a bit late. And yeah checking db, sqlalchemy is a good debug.

**Comment by moadennagi** [comment]
The issue disappears when removing the call to `db.commit()`,  or when adding `db.refresh`
